CODE/Sources/PennyBOT/+Array.swift

```diff
@@ -0,0 +1,40 @@
+
+extension Array {
+    func divided(
+        _ isInLhs: (Element) async throws -> Bool
+    ) async rethrows -> (lhs: Array<Element>, rhs: Array<Element>) {
+        var lhs = ContiguousArray<Element>()
+        var rhs = ContiguousArray<Element>()
+
+        var iterator = self.makeIterator()
+
+        while let element = iterator.next() {
+            if try await isInLhs(element) {
+                lhs.append(element)
+            } else {
+                rhs.append(element)
+            }
+        }
+
+        return (Array(lhs), Array(rhs))
+    }
+
+    func divided(
+        _ isInLhs: (Element) throws -> Bool
+    ) rethrows -> (lhs: Array<Element>, rhs: Array<Element>) {
+        var lhs = ContiguousArray<Element>()
+        var rhs = ContiguousArray<Element>()
+
+        var iterator = self.makeIterator()
+
+        while let element = iterator.next() {
+            if try isInLhs(element) {
+                lhs.append(element)
+            } else {
+                rhs.append(element)
+            }
+        }
+
+        return (Array(lhs), Array(rhs))
+    }
+}
```

Comment:
```suggestion
    func divided(
        _ isInLhs: (Element) throws -> Bool
    ) rethrows -> (lhs: ArraySlice<Element>, rhs: ArraySlice<Element>) {
        var copy = self
        let firstOfRhs = try copy.partition(by: isInLhs)
        return (copy[copy.startIndex ..< firstOfRhs], copy[firstOfRhs ..< copy.endIndex])
    }
}
```

Reply:
thanks!

Reply:
I think this one is causing some problems 👀 tests crashing now